llvm/test/Transforms/Reassociate/fast-basictest.ll

```diff
@@ -578,3 +578,61 @@ define float @test18(float %A, float %B) {
   %Z = fadd fast float %Y, 1.200000e+01
   ret float %Z
 }
+
+; Do not fneg fast float unless it has nnan
+
+define float @scalar(float %a) {
+; CHECK-LABEL: @scalar(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[NEG:%.*]] = fmul fast float [[A:%.*]], 2.000000e+00
+; CHECK-NEXT:    [[MUL:%.*]] = fmul fast float [[NEG]], [[A]]
+; CHECK-NEXT:    ret float [[MUL]]
+;
+entry:
+  %b = fmul fast float %a, -2.0
+  %neg = fneg fast float %a
+  %mul = fmul fast float %neg, %b
```

Comment:
Don't need the full set of fast flags. Also don't need it on every instruction. Should target testing missing specific flags contexts